src/util/misc.py

```diff
@@ -510,47 +510,51 @@ def visit_UnaryOp(self, node):
                 return self._replace_node("_upg_UnaryOp_Not", node) if isinstance(node.op, ast.Not) else node
 
         replacer = RewriteName()
-        root = ast.parse(expr.strip(), mode="eval").body
+        root = ast.parse(expr.strip(), mode="exec")
         visited = replacer.visit(root)
         return (ast_unparse(visited).strip(), SelfPrintEvalContext(replacer.replaces))
 
 
+GLOB_LIT = object()
+
+
 class _Replacer(ast.NodeTransformer):
     """Replace literal nodes in an AST."""
 
     def __init__(self, mapping):
         self.mapping = collections.defaultdict(list)
         for key, value in mapping.items():
-            key_ast = ast.parse(key, mode="eval").body
+            key_ast = key if isinstance(key, ast.AST) else ast.parse(key, mode="eval").body
             self.mapping[key_ast.__class__.__name__].append((key_ast, value))
 
     def _no_match(self, left, right):
         return False
 
     def _match(self, left, right):
-        same_ctx = getattr(left, "ctx", None).__class__ is getattr(right, "ctx", None).__class__
+        same_ctx = getattr(left, "ctx", None).__class__ is getattr(right, "ctx", ast.Load()).__class__
+        same_ctx = True
```

Comment:
Is this diff wanted?

Reply:
No, it's still a wip detail. Lemme check it once more.